lib/srv/desktop/rdp/rdpclient/src/rdpdr.rs

```diff
@@ -2277,6 +2321,13 @@ impl Encode for DeviceControlResponse {
     }
 }
 
+impl PartialEq for DeviceControlResponse {
+    fn eq(&self, other: &Self) -> bool {
+        self.header == other.header
+            && self.output_buffer.encode().unwrap() == other.output_buffer.encode().unwrap()
```

Comment:
Are the encode calls guaranteed to succeed here?

Reply:
Technically no although it's unlikely. I double checked though, and it turns out that all the `PartialEq`s I added are unnecessary because the `ContextInternal`'s `PartialEq` was unnecessary, so I went ahead and removed all of them:

8a21a0b9461b8f7788a1c205df38675d4a74c3b6